tlair/src/test/scala/at/forsyte/apalache/tla/lir/io/TestJsonWriter.scala

```diff
@@ -0,0 +1,475 @@
+    // Real
+    compare(
+      ValEx(TlaRealSet), // TODO: builders for sets? (Andrey)
+      """"set:Real""""
+    )
+  }
+
+  test("prime name") {
+    // awesome'
+    compare(
+      prime("awesome"),
+      """{"prime":"awesome"}"""
+    )
+  }
+
+  test("empty set") {
+    // { }
+    compare(
+      enumSet(),
+      """{"enum":[]}"""
+    )
+  }
+  //
+  test("singleton set") {
+    // { 42 }
+    compare(
+      enumSet(42),
+      """{"enum":[42]}"""
+    )
+  }
+
+  test("singleton set multi-line") {
+    // { 42 }
+    compareMultiLine(
+      enumSet(42),
+      """{
+        |  "enum": [
+        |    42
+        |  ]
+        |}""".stripMargin
+    )
+  }
+
+  test("enum") {
+    // { 1, 2, 3 }
+    compare(
+      enumSet(int(1), int(2), int(3)),
+      """{"enum":[1,2,3]}"""
+    )
+  }
+
+  test("enum multi-line") {
+    // { 1, 2, 3 }
+    compareMultiLine(
+      enumSet(int(1), int(2), int(3)),
+      """{
+        |  "enum": [
+        |    1,
+        |    2,
+        |    3
+        |  ]
+        |}""".stripMargin
+    )
+  }
+
+  test("tuple") {
+    // << 1, two, "three" >>
+    compare(
+      tuple(int(1), name("two"), str("three")),
+      """{"tuple":[1,"two","str:three"]}"""
+    )
+  }
+
+  test("conjunction") {
+    // a /\ b /\ c
+    compare(
+      and(name("a"), name("b"), name("c")),
+      """{"and":["a","b","c"]}"""
+    )
+  }
+
+  test("minus") {
+    // 1 - 2
+    compare(
+      minus(int(1), int(2)),
+      """{"-":[1,2]}"""
+    )
+  }
+
+  test("function definition") {
+    // [ x \in S, y \in T |-> x + y ]
+    compareMultiLine(
+      funDef(plus("x", "y"), "x", "S", "y", "T"),
+      """{
+        |  "function": {
+        |    "+": [
+        |      "x",
+        |      "y"
+        |    ]
+        |  },
+        |  "where": [
+        |    "x",
+        |    "S",
+        |    "y",
+        |    "T"
+        |  ]
+        |}""".stripMargin
+    )
+  }
+
+  test("function application") {
+    // f[e]
+    compare(
+      appFun("f", "e"),
+      """{"apply":"f","to":"e"}"""
+    )
+  }
+
+  test("operator application") {
+    // A(1,2)
+    compare(
+      OperEx(TlaOper.apply, "A", 1, 2),
+      """{"eval":"A","args":[1,2]}"""
+    )
+  }
+
+  test("double function application") {
+    // f[e][g]
+    compare(
+      appFun(appFun("f", "e"), "g"),
+      """{"apply":{"apply":"f","to":"e"},"to":"g"}"""
+    )
+  }
+
+  test("function except") {
+    // [f EXCEPT ![k] = v]
+    compare(
+      except("f", "k", "v"),
+      """{"except":"f","where":["k","v"]}"""
+    )
+  }
+
+  test("record / function enumeration") {
+    // [ k_1 |-> v_1, ..., k_n |-> v_n ]
+    compare(
+      enumFun(
+        str("x1"), "y1",
+        str("x2"), "y2",
+        str("x3"), "y3"
+      ),
+      """{"record":["str:x1","y1","str:x2","y2","str:x3","y3"]}"""
+    )
+  }
+
+  test("function set") {
+    // [S -> T]
+    compare(
+      funSet(name("S"), name("T")),
+      """{"function-set":["S","T"]}"""
```

Comment:
alternatively, we could use shorter names, like `funset`, `recset`.

Reply:
changed to fun-set and rec-set